src/main/scala/dxWDL/compiler/Tree.scala

```diff
@@ -57,35 +56,75 @@ case class Tree(execDict: Map[String, ExecRecord]) {
     }
   }
 
-  def prettyPrint(primary: Native.ExecRecord, indent: Int = 0): String = {
-    primary.callable match {
-      case apl: IR.Applet if apl.kind.isInstanceOf[AppletKindTask] =>
-        Utils.genNSpaces(indent) + Console.GREEN + apl.name + Console.RESET
+  // Traverse the exec tree and generate an appropriate name + color based on the node type
+  // pass back the prefix for the next node build on.
+  def prettyPrint(primary: Native.ExecRecord,
+                  stageDesc: Option[String] = None,
+                  indent: Int = 3,
+                  prefix: String = ""): String = {
 
-      case apl: IR.Applet if apl.kind.isInstanceOf[AppletKindWfFragment] =>
-        // applet that calls other applets/workflows at runtime.
-        // recursively describe all called elements.
-        val links: Vector[String] = primary.links.map {
-          case eli =>
-            val calleeRecord = execDict(eli.name)
-            prettyPrint(calleeRecord, indent + 4)
-        }.toVector
-        val topLine = Utils.genNSpaces(indent) + Console.CYAN + apl.name + Console.RESET
-        val lines = topLine +: links
-        lines.mkString("\n")
+    val lastElem = s"└${"─" * indent}"
+    val midElem = s"├${"─" * indent}"
 
-      case wf: IR.Workflow =>
-        val stageLines: Vector[String] = wf.stages.map {
-          case stage =>
-            val calleeRecord = execDict(stage.calleeName)
-            val calleeDesc: String = prettyPrint(calleeRecord, indent + 4)
-            val topLine = Utils
-              .genNSpaces(indent) + Console.MAGENTA + stage.description + Console.RESET
-            topLine + calleeDesc
+    primary.callable match {
```

Comment:
Can you please add documentation so that it would be understandable to somebody reading the code, or wanting to modify it? 
A good way to do it, in this case, is to give a working example, and explain what the different parts in the algorithm are aiming to do.

Reply:
I added comments, as well as a walk through in latest commit, let me know how that looks and I can resolve this.